pyflakes/test/test_type_annotations.py

```diff
@@ -449,6 +449,31 @@ def f() -> Optional['Queue[str]']:
             return None
         """)
 
+    @skipIf(version_info < (3,), 'new in Python 3')
+    def test_partially_quoted_type_assignment(self):
+        self.flakes("""
+        from queue import Queue
+        from typing import Optional
+
+        MaybeQueue = Optional['Queue[str]']
+        """)
+
+    @skipIf(version_info < (3,), 'new in Python 3')
+    def test_quoted_type_cast(self):
+        self.flakes("""
+        from typing import cast, Optional
+
+        maybe_int = cast('Optional[int]', 42)
+        """)
+
+    @skipIf(version_info < (3,), 'new in Python 3')
+    def test_quoted_type_cast_renamed_import(self):
+        self.flakes("""
+        from typing import cast as tsac, Optional as Maybe
+
+        maybe_int = tsac('Maybe[int]', 42)
+        """)
```

Comment:
all three of these tests should pass in python2.x (?)

Reply:
Ah yeah. I'm too used to working Python 3 only.

Reply:
same lol <3

Reply:
Done (with rebasing).

Meta: what's your (this repo/organisation)'s preference on rebasing versus adding commits to the end of the branch?

Reply:
that's hmmm a good question -- at the end we do merge commits but I think it's up to the author to tidy commits / etc.